src/baseUser/baseUser.controller.ts

```diff
@@ -0,0 +1,41 @@
+import {Body, Controller, Get, Param, Post, Put, Req} from '@nestjs/common';
+import {BaseUserService} from './baseUser.service';
+import {RegisterUserDto} from './dto/registerUser.dto';
+import {LoginUserDto} from './dto/loginUser.dto';
+import {UpdateUserDto} from './dto/updateUser.dto';
+import {SearchUsersDto} from './dto/searchUsers.dto';
+import {Public} from 'src/public.decorator';
+import {CustomRequest} from './customRequest.type';
+
+@Controller('users')
+export class BaseUserController {
+    constructor(private readonly baseUserService: BaseUserService) {}
+
+    @Get(':id')
+    getUser(@Param('id') id: string) {
+        return this.baseUserService.getUser(id);
+    }
+
+    @Post('search')
+    search(@Body() searchUserDto: SearchUsersDto) {
+        return this.baseUserService.search(searchUserDto);
+    }
+
+    @Public()
+    @Post('register')
+    register(@Body() registerUserDto: RegisterUserDto) {
+        return this.baseUserService.registerUser(registerUserDto);
+    }
+
+    @Public()
+    @Post('login')
+    login(@Body() loginUserDto: LoginUserDto) {
+        return this.baseUserService.login(loginUserDto);
+    }
+
+    @Put(':id')
+    update(@Param('id') id: string, @Body() updateUserDto: UpdateUserDto, @Req() request: CustomRequest) {
+        const requestUserId = request.user.id;
```

Comment:
btw, interesting QoL task: try to implement a decorator, which can extract user data from request and can be used like
`someMethod(@UserData() user: UserData)`